From 41e5199f0e4a3e9d518d6f95a3efdfae5e2c7913 Mon Sep 17 00:00:00 2001
From: Bruno Haible <bruno@clisp.org>
Date: Mon, 20 Mar 2017 22:36:36 +0100
Subject: [PATCH] Fix compilation error in <libintl.h> on Cygwin.

Reported by Michael Haubenwallner at https://savannah.gnu.org/bugs/?50595 .

* gettext-runtime/intl/libgnuintl.in.h (newlocale): Hide declaration if
  locale_t type is not visible on Cygwin.
---
 gettext-runtime/intl/libgnuintl.in.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gettext-runtime/intl/libgnuintl.in.h b/gettext-runtime/intl/libgnuintl.in.h
index dc1947f..95dcc16 100644
--- a/gettext-runtime/intl/libgnuintl.in.h
+++ b/gettext-runtime/intl/libgnuintl.in.h
@@ -439,7 +439,10 @@ extern char *setlocale (int, const char *);
 
 #undef newlocale
 #define newlocale libintl_newlocale
+/* Declare newlocale() only if the system headers define the 'locale_t' type. */
+#if !(defined __CYGWIN__ && !defined LC_ALL_MASK)
 extern locale_t newlocale (int, const char *, locale_t);
+#endif
 
 #endif
 
-- 
1.9.1

